src/content/blog/rust-based-javascript-linters-are-feature-incomplete-right-now/index.mdx

```diff
@@ -0,0 +1,130 @@
+Without type information, they wouldn't be able to understand the type of any value imported from another module.
+
+### Type-Checked Linting Performance
+
+The main downside of type-checked linting is performance.
+Lint rules are now blocked on calling to an API such as TypeScript's, which generally need to read all files to see which ones impact types of any other file.
+That means linting performance will often be strictly worse than that of running `tsc` on your entire project.
+
+We're actively working on this in typescript-eslint.
+Our [Performance Troubleshooting](https://typescript-eslint.io/linting/troubleshooting/performance-troubleshooting) docs have some suggestions, and we're very hopeful that our [`EXPERIMENTAL_useProjectService` option](https://typescript-eslint.io/packages/parser/#experimental_useprojectservice) will land as stable in 2024.
+But even if that option works perfectly then type-checked linting will by design be orders of magnitude slower than traditional linting.
+
+Our experience has been that the benefits of type-checked linting almost always far outweigh the drawbacks of slower performance.
+Most of the time, when we've seen projects with slow type-checked linting, the root cause was either a misconfiguration of typescript-eslint or [slow TypeScript types](https://github.com/microsoft/TypeScript/wiki/Performance).
+
+## Rust-Based Linters and Type Checking
+
+**No Rust-based linter has integrated with TypeScript's type checking APIs yet**.
+That means no Rust-based linter is a full replacement for ESLint + typescript-eslint.
+
+I'm not saying you shouldn't use a Rust-based linter: if you don't want any of the type-checked lint rules, then sure, switching over is great.
+But I strongly recommend you look through at least the [recommended type-checked rules in typescript-eslint](https://typescript-eslint.io/rules/?=recommended-typeInformation) to understand what you're missing first.
+
+### Integrating Type-Checked Linting and Rust-Based Linters
+
+Right now, TypeScript is the only tool that can reliably provide type-checked linting for TypeScript code.
+TypeScript is written in TypeScript, so its type checking runs at JavaScript speed.
+
+In order to work with type checking, a Rust linter would have to either:
+
+- Take the performance hit of calling to TypeScript's JavaScript-speed type-checking APIs
+- Reimplement TypeScript's APIs in a native speed language
+- Speed Up TypeScript's APIs to native speed
+
+#### Option: Slowing Down to JavaScript Speed
+
+The performance hit option would likely slow the linters down to the point where they have no noticeable performance advantage compared to ESLint.
+👎
+
+#### Option: Reimplementing TypeScript at Native Speed
+
+Reimplementing TypeScript at native speed is a tantalizing prospect for TypeScript users in general, not just linters.
+I know of two projects that are working on it: [Ezno](https://github.com/kaleidawave/ezno) and [`stc`](https://stc.dudy.dev).
```

Comment:
[Todo] Similar to #140's reaching-outs, I'll reach out to the teams of each of these to make sure I'm not misrepresenting anything.